docs/products/kafka/howto/ksql-docker.rst

```diff
@@ -0,0 +1,106 @@
+
+* Use :doc:`Aiven for Apache Flink </docs/products/flink/index>` 
+* Run a self-hosted ksqlDB cluster.
+
+
+Prerequisites
+-------------
+
+To connect ksqlDB to Aiven for Apache Kafka® you need to create a :doc:`Java keystore and truststore containing the service SSL certificates <keystore-truststore>`. 
+
+Furthermore, you need to collect the following information:
+
+* ``APACHE_KAFKA_HOST``: The Aiven for Apache Kafka hostname
+* ``APACHE_KAFKA_PORT``: The Aiven for Apache Kafka port
+* ``SCHEMA_REGISTRY_PORT``: The Aiven for Apache Kafka schema registry port, if enabled
+* ``SCHEMA_REGISTRY_PASSWORD``: The password associated to the ``avnadmin`` user for Schema Registry
+* ``KEYSTORE_FILE_NAME``: The name of the Java keystore containing the Aiven for Apache Kafka SSL certificates
+* ``TRUSTSTORE_FILE_NAME``: The name of the Java truststore containing the Aiven for Apache Kafka SSL certificates
+* ``SSL_KEYSTORE_PASSWORD``: The password used to secure the Java keystore
+* ``SSL_KEY_PASSWORD``: The password used to secure the Java key
+* ``SSL_TRUSTSTORE_PASSWORD``: The password used to secure the Java truststore
+* ``SSL_STORE_FOLDER``: The absolute path of the folder containing both the truststore and keystore
+* ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME``: The name of the Java truststore containing the schema registry (`Karapace <https://karapace.io/>`__) certificate 
+* ``TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD``: The password used to secure the Java truststore for the schema registry (`Karapace <https://karapace.io/>`__)  certificate
+
+Create a keystore for schema registry's ca file
+'''''''''''''''''''''''''''''''''''''''''''''''
+
+ksqlDB by default uses the ``ssl.truststore`` settings for the Schema Registry connection. 
+
+To have ksqlDB working with Aiven's `Karapace <https://karapace.io/>`__ Schema Registry you need to explicitly define a truststore that contains the commonly trusted root CA of Schema Registry server. To create such a truststore:
+
+1. Obtain the root CA of the server with the following ``openssl`` command by replacing the ``APACHE_KAFKA_HOST`` and ``SCHEMA_REGISTRY_PORT`` placeholders::
+
+    openssl s_client -connect APACHE_KAFKA_HOST:SCHEMA_REGISTRY_PORT \
+        -showcerts < /dev/null 2>/dev/null | \
+        awk '/BEGIN CERT/{s=1}; s{t=t "\n" $0};
+            /END CERT/ {last=t; t=""; s=0}; END{print last}' \
+        > ca_schema_registry.cert
+
+2. Create the truststore with the following ``keytool`` command  by replacing the ``TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME`` placeholder::
+
+    keytool -import -file ca_schema_registry.cert \
+        -alias CA \
+        -keystore \
+        TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME
+   
+   You'll be prompted to insert a password, which in this example is referenced by the placeholder ``TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD``
```

Comment:
Password prompt can be skipped by adding `-noprompt` and `-storepass` options provided.

    keytool -import -file ca_schema_registry.cert \
        -alias CA \
        -keystore TRUSTSTORE_SCHEMA_REGISTRY_FILE_NAME \
        -storepass TRUSTSTORE_SCHEMA_REGISTRY_PASSWORD \
        -noprompt

Reply:
Right thanks!